ambf_utilities/tf_utils/scripts/frame.py

```diff
@@ -0,0 +1,200 @@
+""" This class will generate a Frame in 3D space to mimic Frame in pyKDL"""
+
+import numpy as np
+from numpy import pi, sin, cos, tan, arctan2
+from vector import Vector
+from twist import Twist
+from wrench import Wrench
+from rotation import Rotation
+from warnings import warn
+import sys
+
+
+class Frame(object):
+
+  M = Rotation()
+  p = Vector()
+
+  @staticmethod
+  def make_Frame(HTM):
+    """
+    Return a frame from a 4x4 Homogenous transformaiton matrix
+
+    @param 4x4mat Homogenous TF Matrix
+
+    @return Frame
+    """
+
+    frame = Frame(HTM[:3, :3], HTM[0:3, 3])
+    return frame
+
+  @staticmethod
+  def make_HTM(f):
+    """
+    Return a 4x4 Homogenous Transformation Matrix from a Frame
+
+    @param Frame f
+    
+    @return 4x4mat Homogenous TF Matrix
+    """
+    
+    mat_44 = np.ndarray((4, 4))
+    mat_44[:3, :3] = f.M
+    mat_44[3:, :4] = np.array([0, 0, 0, 1])
+    mat_44[:3, 3] = f.p
+
+    return mat_44
+
+  @staticmethod
+  def Identity():
+    """
+        Constructs an identity frame
+
+        @return Frame
+        """
+
+    return Frame()
+
+  @staticmethod
+  def HD(a, alpha, d, theta):
```

Comment:
What is this called HD? Should it not be called DH?

Reply:
Yes, silly mistake, will update it.